ckan/logic/__init__.py

```diff
@@ -578,8 +578,8 @@ def my_custom_action_function(context, data_dict):
 
     '''
     @functools.wraps(action)
-    def wrapper(context, data_dict):
-        return action(context, data_dict)
+    def wrapper(*args, **kwargs):
+        return action(*args, **kwargs)
     wrapper.side_effect_free = True
```

Comment:
there's no need for the wrapper here at all, right? How about:
```python
def side_effect_free(action):
    '...'
    action.side_effect_free = True
    return action
```

Reply:
that seems better - makes it much more like the chained_action definition as well.